### DISCUSSION TITLE
compatible with C89?

### DISCUSSION THREAD

**Discussion by YuGiOhJCJ** [recp/cglm#222] [Q&A]
Hello,
In the [cglm documentation](https://cglm.readthedocs.io/) I can read:
> compatible with C89

However, when I try to compile this code:
```
#include <cglm/affine.h> /* for glm_translate */
#include <cglm/mat4.h> /* for GLM_MAT4_IDENTITY */
int main()
{
	/* declare */
	mat4 trans = GLM_MAT4_IDENTITY_INIT;
	/* transform */
	glm_translate(trans, (vec3){0.5, -0.5, 0});
	glm_rotate(trans, glm_rad(90), (vec3){0, 0, 1});
	return 0;
}
```
I get:
```
$ gcc -std=c89 -pedantic -Wall -Werror -g -o main.out main.c  -lm
In file included from /usr/include/cglm/simd/intrin.h:83,
                 from /usr/include/cglm/common.h:43,
                 from /usr/include/cglm/affine.h:36,
                 from main.c:1:
/usr/include/cglm/simd/x86.h:57:8: error: unknown type name ‘inline’
 static inline
        ^~~~~~
[...]
```
So, the `inline` keyword is not part of C89.
This is confirmed in the [GSL documentation](https://www.gnu.org/software/gsl/doc/html/usage.html#inline-functions):
> The inline keyword is not part of the original ANSI C standard (C89) 

Of course, the code above can be compiled with C99:
```
$ gcc -std=c99 -pedantic -Wall -Werror -g -o main.out main.c  -lm
```
Is there a way to compile with C89 please?
Thank you.
Best regards.

**Comment by recp** [comment]
@YuGiOhJCJ I'm sorry, I didn't see this question, sorry again,

The `inline` functionality can be provided to C89 via compiler extensions, but we could provide better way to build **cglm** with C89, I added this to my and **cglm**'s TODOs

**Reply by YuGiOhJCJ** [reply]
Indeed for better code portability I am using C89 without compiler specific extensions.
This could be a nice improvement for **cglm** to provide support for that.
Thanks for adding that to the **cglm**'s TODOs.

**Reply by recp** [reply]
@YuGiOhJCJ thanks, if you or community have any idea what to do about this, it would be nice to hear :) Maybe we could disable `inline` and make all functions just static in headers; `static inline` -> `static` (by configuring `CGLM_INLINE` macro)? Because it would be nice to not change existig design too much in favor of `inline` if it is possible
